kmk/extensions/power.py

```diff
@@ -0,0 +1,93 @@
+from kmk.extensions import Extension
+from kmk.kmktime import sleep_ms, ticks_diff, ticks_ms
+
+
+class Power(Extension):
+    def __init__(self, powersave_pin=None, enable=False, is_target=True):
+        self.enable = enable
+        self.powersave_pin = powersave_pin
+        self.is_target = is_target
+        self._powersave_start = ticks_ms()
+        self._usb_last_scan = ticks_ms() - 5000
+        self._psp = None
+
+    def __repr__(self):
+        return f'Power({self._to_dict()})'
+
+    def _to_dict(self):
+        return f'Power( enable={self.enable} powersave_pin={self.powersave_pin} is_target={self.is_target} _powersave_start={self._powersave_start} _usb_last_scan={self._usb_last_scan} _psp={self._psp} )'
+
+    def during_bootup(self, keyboard):
+        self.enable = not bool(self.usb_scan)
+
+    def before_matrix_scan(self, keyboard_state):
+        if self.usb_rescan_timer():
+            self.enable = not bool(self.usb_scan)
+
+    def after_matrix_scan(self, keyboard_state, matrix_update):
+        if matrix_update:
+            self.psave_time_reset()
+        else:
```

Comment:
`elif not self.usb_scan()` - Why sleep when you have "infinite" power? (muahahahahaha)

Reply:
This is the answer that didn't used to exist, but now does. Beat me to finding the answer that I hadn't looked for yet.